app/jobs/bulk_invoice_job.rb

```diff
@@ -41,4 +43,10 @@ def broadcast(filepath, channel)
       ).
       broadcast
   end
+
+  def ensure_directory_exists(filepath)
+    return unless File.exist?(File.dirname(filepath))
+
+    FileUtils.mkdir_p(File.dirname(filepath))
```

Comment:
`mkdir_p` is okay with the directory existing:
```suggestion
    FileUtils.mkdir_p(File.dirname(filepath))
```